ADR/0039-spdx-support.md

```diff
@@ -0,0 +1,151 @@
+| components.properties = [                 | package.annotations = [                     |
+|   {“name”: …, “value”: …}                 |   {..., annotator: <tool>”:jsonencoded”     |
+| ]                                         | ]                                           |
+
+
+#### Formulations
+CycloneDX formulations describe how the container was manufactured. In SPDX, Relationship elements can be used for the same purpose. All elements in SPDX have SPDXID attribute which is an element identifier unique in the whole SBOM document. Relationship element describes relation between two elements using their SPDXID and relationship type. Relationship type BUILD_TOOL_OF can be used to express the relationship of packages which were used to build the container.
+
+| CycloneDX Attribute             | SPDX Attribute                                             |
+|---------------------------------|------------------------------------------------------------|
+| Formulations.components = [{}]  | Relationships = [{                                         |
+|                                 |  spdxElementId = <A-PACKAGE-ID>                            |
+|                                 |  relationshipType=BUILD_TOOL_OF                            |
+|                                 |  relatedSpdxElement=<ROOT-DOCUMENT-ID>                     |
+|                                 | }]                                                         |
+
+
+#### Metadata.tools
+The CycloneDX metadata.tools sub attributes that we are mostly interested in are the vendor and name elements. Information about the creation of the SPDX document can be stored into creationInfo. CreationInfo.creators element is basically a list of strings. There’s a vague specification ([here](https://spdx.github.io/spdx-spec/v2.3/document-creation-information/#68-creator-field)]) about how it should be structured in the standard. Strings should be formatted in the following way: “<Attribute>: <Value>”. For example vendor should be stored as “Vendor: <vendor>”
+
+| CyloneDX Attribute                             | SPDX Attribute                                    |
+|------------------------------------------------|---------------------------------------------------|
+| Metadata.tools = [{“vendor”: “X”, “name”: “Y”] | CreationInfo.creators = [“Vendor: X”, “Tool: Y”]  |
+
+
+#### Merging SPDX
+##### Packages
+Packages of two SPDX documents can be merged together as a concatenation of two lists. In cycloneDX component elements can have only a single purl attribute, therefore component elements representing packages with the same name and version but with different purl have to be stored as multiple elements. SPDX package elements can bear multiple purls. Therefore multiple cycloneDX components can be squashed together into single SPDX package element with purls concatenated into a single list. Following rules are applied to packages merging process:
```

Comment:
Asking to educate myself: what does it mean to have packages with identical names and versions, but different purls?

Reply:
You can have one document with `pkg:maven/org.apache.commons/commons-lang3@3.12.0` and a second document with `pkg:github/apache/commons-lang@3.12.0`. Ie. one package comes from maven repository and the other from GH repository. 
For RHEL this is IMO theoretical option. But it may be likely for containers or layered products.

Reply:
Or you could have `pkg:npm/requests@1.0.0` and `pkg:pypi/requests@1.0.0`. Or any other common library name in any two ecosystems

Reply:
Or even a package with the same name and version in the _same_ ecosystem, which is actually a different package. This is surprisingly common in the Go world, especially in the context of OpenShift where many modules are forks of Kubernetes module. See for example https://github.com/openshift/cluster-api-provider-aws/blob/master/go.mod (note that the module name matches the name of the original module, not of the fork).

Depending on how the SBOM-generation tool handles this case, the purl may either be completely indistinguishable from the original, or may include the fork url as an attribute, or may use the name of the fork instead of the original name